app_flutter/lib/state/flutter_build.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import 'package:cocoon_service/protos.dart' show CommitStatus;
+
+import '../service/cocoon.dart';
+
+/// State for the Flutter Build Dashboard
+class FlutterBuildState extends ChangeNotifier {
+  /// Cocoon backend service that retrieves the data needed for this state.
+  final CocoonService _cocoonService = CocoonService();
+
+  /// How often to query the Cocoon backend for the current build state.
+  final Duration _refreshRate = Duration(seconds: 10);
+
+  /// Timer that calls [_fetchBuildStatusUpdate] on a set interval.
+  @visibleForTesting
+  Timer refreshTimer;
+
+  /// The current status of the commits loaded.
+  List<CommitStatus> statuses = [];
+
+  /// Start a fixed interval loop that fetches build state updates based on [_refreshRate].
+  void startFetchingBuildStateUpdates() async {
+    if (refreshTimer != null) {
+      // There's already an update loop, no need to make another.
+      return;
+    }
+
+    refreshTimer =
+        Timer.periodic(_refreshRate, (t) => _fetchBuildStatusUpdate());
+  }
+
+  /// Request the latest [statuses] from [CocoonService].
+  void _fetchBuildStatusUpdate() async {
+    statuses = await _cocoonService.fetchCommitStatuses();
+    notifyListeners();
+  }
```

Comment:
What happens when `fetchCommitStatuses` returns a HttpException?

Reply:
That's a good point. Do you have any recommendations on how to forward this error to the widget?

My current approach would be `Stream`, but that seems to over engineer what we want to do here. A naive approach would be to just assume this was a one off incident, and don't update statuses.

Reply:
We talked in person and I created https://github.com/flutter/cocoon/issues/451 that sums up our solution.